src/main/java/StationsGraph.java

```diff
@@ -0,0 +1,119 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+public class StationsGraph {
+    private final HashMap<String, StationNode> stationNodes;
+
+    /**
+     * Constructor creates all stationNodes in the graph
+     * @param stations Hashmap
+     */
+    public StationsGraph(HashMap<String, List<String>> stations) {
+        this.stationNodes = new HashMap<>();
+        //for each id of the stations hashmap, it creates a stationNode with that id
+        //and then puts the id and stationNode in the stationNodes hashMap
+        for(String stationId: stations.keySet()){
+            stationNodes.put(stationId, new StationNode(stationId));
+        }
+        //connecting stationNodes to each other
+        //for each id in stations, it stores the neighbors of that stationId in a List
+        for(String stationId : stations.keySet()){
```

Comment:
https://stackoverflow.com/questions/46898/how-do-i-efficiently-iterate-over-each-entry-in-a-java-map